src/main/java/seedu/address/storage/JsonCookbookStorage.java

```diff
@@ -44,7 +45,18 @@ public Path getCookbookFilePath() {
     public Optional<ReadOnlyCookbook> readCookbook(Path filePath) throws DataConversionException {
         requireNonNull(filePath);
 
-        return Optional.empty();
+        Optional<JsonSerializableCookbook> jsonCookbook = JsonUtil.readJsonFile(
+                filePath, JsonSerializableCookbook.class);
+        if (!jsonCookbook.isPresent()) {
+            return Optional.empty();
+        }
+
+        try {
+            return Optional.of(jsonCookbook.get().toModelType());
+        } catch (IllegalValueException ive) {
+            logger.info("Illegal values found in " + filePath + ": " + ive.getMessage());
+            throw new DataConversionException(ive);
+        }
```

Comment:
There's a conflict with this in my PR. Feel free to accept yours when you merge, I think there shouldn't be much difference.